### PR TITLE
Fix for data stores resummarized without changes. (#8990)

### PR DESCRIPTION
**Pull request by agarwal-navin**
Cherry-pick from https://github.com/microsoft/FluidFramework/pull/8990.

Fixes https://github.com/microsoft/FluidFramework/issues/8963.

**Issue**
When a container is loaded, garbage collector reads the GC blobs for data stores from the base snapshot and passes it to data stores during creation. However, it does not sort the used routes before passing it to the data stores. These used routes become referenceUsedRoutes for the data stores, i.e., used routes in the previous summary.
During summarize, the current used routes are compared with the referenceUsedRoutes to determine if the reference state of the data store changed. The used routes must be sorted so that they can be compared correctly.

**Fix**
Data stores sort the used routes passed by the container runtime before saving it as referenceUsedRoutes.

### CODE REVIEW

**Review by vladsud — APPROVED** [review]

**Comment by tylerbutler** [comment]
> @tylerbutler Does this need your approval before it can go in?

No. The only requirement is that you have one review approval. The code owners are asked to review, but you shouldn't be blocked.

**Comment by agarwal-navin** [comment]
I am blocked, it isn't letting me merge the changes:
image

**Review by tylerbutler — APPROVED** [review]

**Comment by tylerbutler** [comment]
@agarwal-navin Weird, sorry about that. I confirmed the settings are such that this shouldn't happen, so please let me know if it does. Could be a GitHub bug.

image

**Comment by agarwal-navin** [comment]
Thanks Tyler. Will let you know if I see this again.

**Pull request merged**
